File: Simplify_Path/Simplify_Path.py
# path = "/home/"
path = "/home//foo/"

# def simplifyPath(path):
#     path = path.split("/")
#     ls = []
#     for pa in path:
#         if pa == "." or pa == "":
#             continue
#         if pa == "..":
#             if ls:
#                 ls.pop()
#         else:
#             ls.append(pa)

#     return "/"+"/".join(ls)


# print(simplifyPath(path))


def simplifyPath(path):
    def rec(path,s1,p1):
        # if path == "":
        #     return s1
        print(len(path),len(s1))
        # for p in p1:
        #     if p == "." or p == "":
        #         continue
        #     if p =="..":
        #         if s1:
        #             s1.pop()
        #     else:
        #         s1.append(p)

        # print(s1)

        

    return rec(path,[],path.split("/"))


print(simplifyPath(path))

File: Simplify_Path/Simplify_Path.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	// path := "/home/"
	path := "/a/./b/../../c/"

	fmt.Println(simplifyPath(path))
}

func simplifyPath(path string) string {
	p1 := strings.Split(path, "/")
	res := make([]string, 0)
	for _, p := range p1 {
		if p == "." || p == "" {
			continue
		}
		if p == ".." {
			if len(res) >= 1 {
				res = res[:len(res)-1]
			}
		} else {
			res = append(res, p)
		}

	}
	return "/" + strings.Join(res, "/")
}
